jdaviz/app.py

```diff
@@ -1570,6 +1573,53 @@ def get_viewer_reference_names(self):
         # Cannot sort because of None
         return [self._viewer_item_by_id(vid).get('reference') for vid in self._viewer_store]
 
+    def update_viewer_reference_name(
+        self, old_reference, new_reference, update_id=False
+    ):
+        """
+        Update viewer reference names.
+
+        Viewer IDs will not be changed unless ``update_id`` is True.
+
+        Parameters
+        ----------
+        old_reference : str
+            The viewer reference name to be changed
+        new_reference : str
+            The viewer reference name to use instead of ``old_reference``
+        update_id : bool, optional
+            If True, update the viewer IDs as well as the viewer reference names.
+        """
+        if new_reference in self.get_viewer_reference_names():
+            raise ValueError(f"viewer with reference='{new_reference}' already exists")
+
+        # update the viewer item's reference name
+        viewer_item = self._get_viewer_item(old_reference)
+        viewer_item['reference'] = new_reference
+
+        if viewer_item['name'] == old_reference:
+            viewer_item['name'] = new_reference
+
+        # optionally update the viewer IDs:
+        if update_id and viewer_item['id'] == old_reference:
```

Comment:
I disagree with changing the ID. ID is like a primary key. It should be static after creation. Reference can be change and we can use that for user renaming as you are doing. But ID should not be touched.